pkg/controller/worker/machines.go

```diff
@@ -324,3 +328,27 @@ func computeAdditionalHashData(pool extensionsv1alpha1.WorkerPool, workerConfig
 
 	return additionalData
 }
+
+func readMachineConfiguration(pool extensionsv1alpha1.WorkerPool) machinev1alpha1.MachineConfiguration {
+	machineConfiguration := machinev1alpha1.MachineConfiguration{}
+	poolSettings := pool.MachineControllerManagerSettings
+	if poolSettings != nil {
+		if poolSettings.MachineDrainTimeout != nil {
+			machineConfiguration.MachineDrainTimeout = poolSettings.MachineDrainTimeout
+		}
+		if poolSettings.MachineHealthTimeout != nil {
+			machineConfiguration.MachineHealthTimeout = poolSettings.MachineHealthTimeout
+		}
+		if poolSettings.MachineCreationTimeout != nil {
+			machineConfiguration.MachineCreationTimeout = poolSettings.MachineCreationTimeout
+		}
+		if poolSettings.MaxEvictRetries != nil {
+			machineConfiguration.MaxEvictRetries = poolSettings.MaxEvictRetries
+		}
+		if len(poolSettings.NodeConditions) > 0 {
+			nodeConditions := strings.Join(poolSettings.NodeConditions[:], ",")
```

Comment:
Any reason for using `[:]`?

```suggestion
			nodeConditions := strings.Join(poolSettings.NodeConditions, ",")
```

Reply:
Nope, slipped through, sorry. Thanks for catching it.